pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/AutoRebalancePartitionAssignmentGenerator.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.controller.helix.core.realtime;
+
+import com.linkedin.pinot.common.config.RoutingConfig;
+import com.linkedin.pinot.common.config.TableConfig;
+import com.linkedin.pinot.common.utils.CommonConstants.Helix.DataSource.RoutingTableBuilderName;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.controller.rebalancer.strategy.AutoRebalanceStrategy;
+
+
+/**
+ * Generates partition assignment for all tables provided, using AutoRebalanceStrategy
+ */
+public class AutoRebalancePartitionAssignmentGenerator extends PartitionAssignmentGenerator {
+
+  public AutoRebalancePartitionAssignmentGenerator(TableConfig tableConfig, int nKafkaPartitions,
+      List<String> instanceNames, Map<String, TableConfig> allTableConfigsInTenant,
+      Map<String, ZNRecord> currentPartitionAssignment) {
```

Comment:
avoid using in the method or completely not use it ?
if in this method, yes, we can pass listFields instead of ZNRecord. But why do we need to avoid ZNRecord?